py-polars/polars/functions/lit.py

```diff
@@ -78,25 +78,42 @@ def lit(
     time_unit: TimeUnit
 
     if isinstance(value, datetime):
+        if dtype == Date:
+            dt_int = date_to_int(value.date())
```

Comment:
Seems like it would be better to stop with the `value.date()` truncation and leave the literal/int conversion to Rust here, rather than inject a cast node, since `lit` can take the Python `date` value as-is. Have you observed any particular advantage to doing this via cast instead?

Reply:
I don't think that `lit` can take a Python date value on the rust side:

```python
from datetime import date
from polars import polars as plr

plr.lit(date(2024, 1, 1), allow_object=False)
```
```
TypeError: cannot create expression literal for value of type date: datetime.date(2024, 1, 1)
```

Reply:
Adding direct date and datetime literal conversion in rust should be relatively simple actually, and I think falls under the purview of this PR.

Reply:
Done, waiting for checks to pass....

Reply:
Good stuff; the addition makes sense (could also have used regular `pl.lit` and graded the `._pyexpr`, but looks like a smart move to handle it at the lower level 👍)